controllers/users.js

```diff
@@ -354,5 +396,65 @@ module.exports = {
         });
       });
     });
+  },
+  updateUser: function(username, data, callback) {
+    var self = this,
+        userID = createID(username),
+        placeID = getDocID(data.place),
+        series = [];
+    if (_.isUndefined(placeID) && _.isUndefined(data.type) && _.isUndefined(data.password)) {
+      return callback(new Error(
+        'One of the following fields are required: type, password or place.'
+      ));
+    }
+    // validate user exists
+    self._getUser(userID, function(err, user) {
+      if (err) {
+        console.error('Failed to find user.');
+        return callback(err);
+      }
+      // validate user settings exist
+      self._getUserSettings(userID, function(err, settings) {
+        if (err) {
+          console.error('Failed to find user settings.');
+          return callback(err);
+        }
+        // update roles
+        if (data.type) {
+          user.roles = data.type ? getRoles(data.type) : user.roles;
+        }
+        // update password
+        if (data.password) {
+          user.password = data.password;
+          series.push(function(cb) {
+            self._updateAdminPassword(username, data.password, cb);
+          });
+        }
+        // validate place exists if it changed
+        if (!_.isUndefined(placeID) && (placeID !== user.facility_id)) {
+          series.push(function(cb) {
+            self._getPlace(placeID, function(err) {
+              if (err) {
+                return cb(err);
+              }
+              user.facility_id = placeID;
+              settings.facility_id = placeID;
+              // update settings
+              self._updateUserSettings(userID, settings, cb);
+            });
+          });
+        // update request only included place but it is the same
+        } else if (_.isUndefined(data.type) && _.isUndefined(data.password)) {
+          return callback(new Error(
```

Comment:
I modified this to just always set (even set the same value) if place passes validation.